### PR TITLE
[process-agent] Separate endpoint for connections payloads

### PR DESCRIPTION
**Pull request by xornivore**
### What does this PR do?

- Use separate endpoint route for connections payload

### Motivation

Being able to route payloads differently by their type.

### Describe how to test/QA your changes

Validate that connections payload submission still works.

### Reviewer's Checklist

- [ ] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [ ] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [ ] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [ ] At least one `team/..` label has been applied, indicating the team(s) that should QA this change.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the `k8s/<min-version>` label, indicating the lowest Kubernetes version compatible with this feature. 
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

**Review by robertjli — APPROVED** [review]
As discussed, this also requires a change on the HAProxy side

**Review by wiyu — APPROVED** [review]

**Review by mbotarro — APPROVED** [review]

File: cmd/process-agent/collector_api_test.go [hunk 1]
```diff
@@ -582,6 +582,7 @@ func newMockEndpoint(t *testing.T, config *endpointConfig) *mockEndpoint {
 	collectorMux := http.NewServeMux()
 	collectorMux.HandleFunc("/api/v1/validate", m.handleValidate)
 	collectorMux.HandleFunc("/api/v1/collector", m.handle)
```
Comment (remeh) [inline cmd/process-agent/collector_api_test.go:584]: Does that still make sense to keep that one in the unit test while you're not using it anymore? Just a confidence check, I did not dig into the unit tests

**Review by remeh — APPROVED** [review]
Left a small comment but LGTM.

File: cmd/process-agent/collector_api_test.go (hunk 1 above)
Comment (xornivore) [reply cmd/process-agent/collector_api_test.go:584]: Yes, thank you, it is still used.

**Pull request merged**